tests/unit/test_spaces.cpp

```diff
@@ -135,28 +135,37 @@ TEST_P(FP32SpacesOptimizationTest, FP32L2SqrTest) {
 
     dist_func_t<float> arch_opt_func;
     float baseline = FP32_L2Sqr(v, v2, dim);
-#ifdef OPT_AVX512F
-    if (optimization.features.avx512f) {
-        arch_opt_func = Choose_FP32_L2_implementation_AVX512(dim);
+    if (optimization.avx512f) {
+        arch_opt_func = L2_FP32_GetDistFunc(dim, &optimization);
+        ASSERT_EQ(arch_opt_func, Choose_FP32_L2_implementation_AVX512(dim))
+            << "Unexpected distance function chosen for dim " << dim;
```

Comment:
Why not to mention for which arch we failed?

Reply:
It can be inferred from the line number